server/src/api/gitLab/gitLabIntegration.ts

```diff
@@ -0,0 +1,213 @@
+    tokens: { accessToken: string },
+  ) {
+    this.projectId = projectId || '';
+    this.lastUsedRedirectUri = '';
+    this.accessToken = tokens.accessToken;
+    this.oauth2 = oauth2(consumerkey, privatekey, host);
+  }
+
+  setLastUsedRedirectUri(uri: string) {
+    this.lastUsedRedirectUri = uri;
+  }
+  getLastUsedRedirectUri() {
+    return this.lastUsedRedirectUri;
+  }
+
+  async boards() {
+    const response = await this.fetch<{ id: number; name: string }[]>('boards');
+    return (response as any[]).map((board) => ({
+      id: `${board.id}`,
+      name: board.name,
+    }));
+  }
+
+  // In GitLab, columns ("lists") can only be created out of labels.
+  // The columns themselves do have an id, but they are named after the label.
+  // Therefore, this returns the column id and the label's name.
+  async columns(boardId: string) {
+    const lists = await this.fetch<
+      Array<{ id: number; label: { name: string } }>
+    >(`boards/${boardId}/lists`);
+    return lists.map((list) => ({ id: `${list.id}`, name: list.label.name }));
+  }
+
+  async labels(boardId: string) {
+    const lists = await this.fetch<Array<{ label: { name: string } }>>(
+      `boards/${boardId}/lists`,
+    );
+    return lists.map((list) => list.label.name);
+  }
+
+  // Inefficient, as GitLab offers no method to get tasks for a given board.
+  // We have to get columns (=labels) for a board, and then get issues for the labels.
+  // This also has to be done one by one, as /issues?labels= filter is AND only.
+  async tasks(boardId: string, filters?: TaskFilters) {
+    const boardColumns = await this.columns(boardId);
+
+    const issues: { [key: string]: string }[] = (
+      await Promise.all(
+        boardColumns.map(async (column) => {
+          const issuesForColumn = await this.getIssuesForLabel(column.name);
+          return issuesForColumn.map((issue) => ({
+            ...issue,
+            columnId: column.id,
+          }));
+        }),
+      )
+    ).flat();
+
+    return issues.filter(this.importFilter(filters)).map(
+      (issue): ImportedTask => ({
+        id: `${issue.id}`,
+        link: issue.web_url,
+        name: issue.title,
+        description: issue.description || 'No description',
+        createdAt: issue.created_at,
+        columnId: `${issue.columnId}`,
+      }),
+    );
+  }
+
+  private fetch<T>(resource: string): Promise<T> {
+    return new Promise((resolve, reject) => {
+      this.oauth2.get(
+        `https://gitlab.com/api/v4/projects/${this.projectId}/${resource}`,
+        this.accessToken,
+        (err, result) => {
+          if (!err) return resolve(JSON.parse(result as string));
+          if (err.statusCode === 401 && /token/i.test(err.data))
+            reject(new InvalidTokenError(err.data));
+          else if (err.statusCode === 400 && /grant/i.test(err.data))
+            reject(new InvalidGrantError(err.data));
+          else reject(err);
```

Comment:
```suggestion
          if (err) reject(convertError(err));
          else resolve(JSON.parse(result as string));
```

Could keep it consistent

This is more obvious when the error check is refactored